### ISSUE TITLE
IMDB Search finds only 100 movies if >1000 results

### ISSUE THREAD

**Issue by BlueSkillZ99** [mza921/Plex-Auto-Collections#159]
Script seems to find only the first 100 movies from IMDB searches, if there are more than 1,000 results. I suspect it might not deal properly with the thousand separator e.g.

`imdb_list: https://www.imdb.com/search/title/?title_type=feature&groups=top_1000&languages=en&count=250`

correctly returns all 800+ results. But

`imdb_list: https://www.imdb.com/search/title/?title_type=feature&groups=top_1000&count=250`

incorrectly returns only 100 (of 1,000) results.

What I am really trying to do is to cap the number of results to 250 or 500. It would be nice to have a manual way to Limit the number of Imbd results

**Comment by jalarab** [comment]
same issue for me. A list with 1082 items but only find first 100

**Comment by BlueSkillZ99** [comment]
I played around a bit and found a workaround. Don't laugh, I do not really know what I am doing and have only minimal coding experience.

Modified imdb_tools.py row 24-57 to 
- Read the number of total IMDB search results correctly, if they include 1 thousand separator 
- Set a global cap to search results of 1500
- Option to set a manual cap for an individual search by extending a search URL with "&maxresult500" (or any other number. Note it rounds up to the next hundred) 

```
def imdb_get_ids(plex, imdb_url):
    imdb_url = imdb_url.strip()
    if imdb_url.startswith("https://www.imdb.com/list/ls") or imdb_url.startswith("https://www.imdb.com/search/title/?"):
        try:
            cap = 1500
            if imdb_url.startswith("https://www.imdb.com/list/ls"):
                imdb_url = "https://www.imdb.com/list/ls{}".format(re.search('(\\d+)', str(imdb_url)).group(1))
            else:
                if "&maxresult" in imdb_url:
                    cap = re.search(r'(?<=maxresult)\d+', imdb_url).group(0)
                    imdb_url = re.sub("&maxresult\d+", "", imdb_url)
                if "&start=" in imdb_url:
                    imdb_url = re.sub("&start=\d+", "", imdb_url)
                if "&count=" in imdb_url:
                    imdb_url = re.sub("&count=\d+", "&count=100", imdb_url)
                else:
                    imdb_url += "&count=100"
            library_language = plex.Library.language
            r = requests.get(imdb_url, headers={'Accept-Language': library_language})
            tree = html.fromstring(r.content)
            title_ids = tree.xpath("//div[contains(@class, 'lister-item-image')]"
                                   "//a/img//@data-tconst")
            if imdb_url.startswith("https://www.imdb.com/list/ls"):
                results = re.search('(?<=<div class="desc lister-total-num-results">).*?(?=</div>)', str(r.content))
                total = 200 if results is None else re.search('(\\d+)', results.group(0).replace(',', '')).group(1)
            else:
                results = re.search('<span>\\d+-\\d+ of \\d+,\\d+ titles.</span>|<span>\\d+-\\d+ of \\d+ titles.</span>', str(r.content))
                temp = 200 if results is None else re.findall('(\\d+,\\d+|\\d+)', results.group(0).replace(',', ''))[2]
                total =  min(int(cap), int(temp))
            for i in range(1, math.ceil(int(total) / 100)):
                if imdb_url.startswith("https://www.imdb.com/list/ls"):
                    r = requests.get(imdb_url + '?page={}'.format(i + 1), headers={'Accept-Language': library_language})
                else:
                    r = requests.get(imdb_url + '&start={}'.format(i * 100 + 1), headers={'Accept-Language': library_language})
                tree = html.fromstring(r.content)
                title_ids.extend(tree.xpath("//div[contains(@class, 'lister-item-image')]"
                                            "//a/img//@data-tconst"))
```

**Comment by jalarab** [comment]
> I played around a bit and found a workaround. Don't laugh, I do not really know what I am doing and have only minimal coding experience.
> 
> Modified imdb_tools.py row 24-55 to
> 
> * Read the number of total IMDB search results correctly, if they include 1 thousand separator
> * Set a global cap to search results of 1500
> * Option to set a manual cap for an individual search by extending a search URL with "&maxresult500" (or any other number. Note it rounds up to the next hundred)
> 
> ```
> def imdb_get_ids(plex, imdb_url):
>     imdb_url = imdb_url.strip()
>     if imdb_url.startswith("https://www.imdb.com/list/ls") or imdb_url.startswith("https://www.imdb.com/search/title/?"):
>         try:
>             cap = 1500
>             if imdb_url.startswith("https://www.imdb.com/list/ls"):
>                 imdb_url = "https://www.imdb.com/list/ls{}".format(re.search('(\\d+)', str(imdb_url)).group(1))
>             else:
>                 if "&maxresult" in imdb_url:
>                     cap = re.search(r'(?<=maxresult)\d+', imdb_url).group(0)
>                     imdb_url = re.sub("&maxresult\d+", "", imdb_url)
>                 if "&start=" in imdb_url:
>                     imdb_url = re.sub("&start=\d+", "", imdb_url)
>                 if "&count=" in imdb_url:
>                     imdb_url = re.sub("&count=\d+", "&count=100", imdb_url)
>                 else:
>                     imdb_url += "&count=100"
>             library_language = plex.Library.language
>             r = requests.get(imdb_url, headers={'Accept-Language': library_language})
>             tree = html.fromstring(r.content)
>             title_ids = tree.xpath("//div[contains(@class, 'lister-item-image')]"
>                                    "//a/img//@data-tconst")
>             if imdb_url.startswith("https://www.imdb.com/list/ls"):
>                 results = re.search('(?<=<div class="desc lister-total-num-results">).*?(?=</div>)', str(r.content))
>                 total = 200 if results is None else re.search('(\\d+)', results.group(0).replace(',', '')).group(1)
>             else:
>                 results = re.search('<span>\\d+-\\d+ of \\d+,\\d+ titles.</span>|<span>\\d+-\\d+ of \\d+ titles.</span>', str(r.content))
>                 temp = 200 if results is None else re.findall('(\\d+,\\d+|\\d+)', results.group(0).replace(',', ''))[2]
>                 total =  min(int(cap), int(temp))
>             for i in range(1, math.ceil(int(total) / 100)):
>                 if imdb_url.startswith("https://www.imdb.com/list/ls"):
>                     r = requests.get(imdb_url + '?page={}'.format(i + 1), headers={'Accept-Language': library_language})
>                 else:
>                     r = requests.get(imdb_url + '&start={}'.format(i * 100 + 1), headers={'Accept-Language': library_language})
>                 tree = html.fromstring(r.content)
>                 title_ids.extend(tree.xpath("//div[contains(@class, 'lister-item-image')]"
>                                             "//a/img//@data-tconst"))
> ```

I tried your workaround but received this error message:

Traceback (most recent call last):
  File "plex_auto_collections.py", line 1026, in <module>
    update_from_config(config_path, plex, True, args.no_meta, args.no_images)
  File "plex_auto_collections.py", line 120, in update_from_config
    plex_map = get_map(config_path, plex)
  File "F:\Plex-Auto-Collections\app\plex_tools.py", line 125, in get_map
    key_id = imdb_tools.imdb_get_tmdb(config_path, check_id)
  File "F:\Plex-Auto-Collections\app\imdb_tools.py", line 66, in imdb_get_tmdb
    movie = Movie()
NameError: name 'Movie' is not defined

**Comment by BlueSkillZ99** [comment]
You are missing Part of the code. Error references row 66. In the latest master „movie = Movie()„ is in row 70. After my modifications, it should be in 75

**Comment by jalarab** [comment]
sorry mi mistake. I checked it and I had no error, but it find only the first 100 items of the list
my problematic list is this https://www.imdb.com/list/ls090936109
maybe you can check

> You are missing Part of the code. Error references row 66. In the latest master „movie = Movie()„ is in row 70. After my modifications, it should be in 75

sorry, my mistake. I checked it and I had no error, but it finds only the first 100 items of the list
my problematic list is this https://www.imdb.com/list/ls090936109
maybe you can check

**Comment by BlueSkillZ99** [comment]
my fix addresses only search URLs, not lists. but you can easily turn any list URL into a search URL. Try this URL instead

https://www.imdb.com/search/title/?lists=ls090936109

**Comment by jalarab** [comment]
> my fix addresses only search URLs, not lists. but you can easily turn any list URL into a search URL. Try this URL instead
> 
> https://www.imdb.com/search/title/?lists=ls090936109

Perfect, now it works, thanks.

**Comment by mza921** [comment]
Is the 1 thousand separator what's breaking this?

**Comment by BlueSkillZ99** [comment]
As I understand, the „\\d+“ regex expression in row 46 and/or 47 does only match regular digits, but not decimal or thousand separators. So I just included second option to match „\\d+,\\d+“

Probably a similar issue with lists in row 43/44

**Comment by mza921** [comment]
I can fix the comma issue, but I want to look into a different way to limit the results other than faking it in the url.

**Comment by BlueSkillZ99** [comment]
Perfect, thank you! Faking  it in the URL is certainly not the best way, but the only one I was capable to implement. Would be nice to  do it similar to „limit“ for tmdb_discover

**Comment by burkasaurusrex** [comment]
> I can fix the comma issue, but I want to look into a different way to limit the results other than faking it in the url.

Yea, I think the right way to do this is detecting the `Next` link and following it instead of trying to emulate the URL logic

**Comment by jalarab** [comment]
> I played around a bit and found a workaround. Don't laugh, I do not really know what I am doing and have only minimal coding experience.
> 
> Modified imdb_tools.py row 24-55 to
> 
> * Read the number of total IMDB search results correctly, if they include 1 thousand separator
> * Set a global cap to search results of 1500
> * Option to set a manual cap for an individual search by extending a search URL with "&maxresult500" (or any other number. Note it rounds up to the next hundred)
> 
> ```
> def imdb_get_ids(plex, imdb_url):
>     imdb_url = imdb_url.strip()
>     if imdb_url.startswith("https://www.imdb.com/list/ls") or imdb_url.startswith("https://www.imdb.com/search/title/?"):
>         try:
>             cap = 1500
>             if imdb_url.startswith("https://www.imdb.com/list/ls"):
>                 imdb_url = "https://www.imdb.com/list/ls{}".format(re.search('(\\d+)', str(imdb_url)).group(1))
>             else:
>                 if "&maxresult" in imdb_url:
>                     cap = re.search(r'(?<=maxresult)\d+', imdb_url).group(0)
>                     imdb_url = re.sub("&maxresult\d+", "", imdb_url)
>                 if "&start=" in imdb_url:
>                     imdb_url = re.sub("&start=\d+", "", imdb_url)
>                 if "&count=" in imdb_url:
>                     imdb_url = re.sub("&count=\d+", "&count=100", imdb_url)
>                 else:
>                     imdb_url += "&count=100"
>             library_language = plex.Library.language
>             r = requests.get(imdb_url, headers={'Accept-Language': library_language})
>             tree = html.fromstring(r.content)
>             title_ids = tree.xpath("//div[contains(@class, 'lister-item-image')]"
>                                    "//a/img//@data-tconst")
>             if imdb_url.startswith("https://www.imdb.com/list/ls"):
>                 results = re.search('(?<=<div class="desc lister-total-num-results">).*?(?=</div>)', str(r.content))
>                 total = 200 if results is None else re.search('(\\d+)', results.group(0).replace(',', '')).group(1)
>             else:
>                 results = re.search('<span>\\d+-\\d+ of \\d+,\\d+ titles.</span>|<span>\\d+-\\d+ of \\d+ titles.</span>', str(r.content))
>                 temp = 200 if results is None else re.findall('(\\d+,\\d+|\\d+)', results.group(0).replace(',', ''))[2]
>                 total =  min(int(cap), int(temp))
>             for i in range(1, math.ceil(int(total) / 100)):
>                 if imdb_url.startswith("https://www.imdb.com/list/ls"):
>                     r = requests.get(imdb_url + '?page={}'.format(i + 1), headers={'Accept-Language': library_language})
>                 else:
>                     r = requests.get(imdb_url + '&start={}'.format(i * 100 + 1), headers={'Accept-Language': library_language})
>                 tree = html.fromstring(r.content)
>                 title_ids.extend(tree.xpath("//div[contains(@class, 'lister-item-image')]"
>                                             "//a/img//@data-tconst"))
> ```

can we update to the new versions just keeping your changes in  imdb_tools.py?

**Comment by mza921** [comment]
The 100 movies bug should already be fixed in 2.9.0.
